### PR TITLE
Create/close loan

### PR DESCRIPTION
**Pull request by evgenyboxer**
- [x] Create loan
- [x] Close loan

Still not hooked to redux, and has no "feedback" when it succeeds... but it works!

### CODE REVIEW

File: src/pages/Loans/components/LoanCards/CloseLoanCard/CloseLoanCard.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+			const gasEstimate = await EtherCollateral.contract.estimate.closeLoan(loanIdStr);
+			const updatedGasEstimate = Number(gasEstimate);
+			setLocalGasLimit(updatedGasEstimate);
+
+			const tx = await EtherCollateral.closeLoan(loanIdStr, {
+				gasPrice: gasInfo.gasPrice * GWEI_UNIT,
+				gasLimit: updatedGasEstimate,
+			});
+
+			console.log(tx);
```
Comment (clementbalestrat) [inline src/pages/Loans/components/LoanCards/CloseLoanCard/CloseLoanCard.js]: maybe remove the console.log? except if you still need it

File: src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+				gasLimit,
+			};
+
+			const gasEstimate = await EtherCollateral.contract.estimate.openLoan(openLoanArgs);
+			const updatedGasEstimate = Number(gasEstimate);
+			setLocalGasLimit(updatedGasEstimate);
+
+			const tx = await EtherCollateral.openLoan({ ...openLoanArgs, gasLimit: updatedGasEstimate });
+
+			console.log(tx);
```
Comment (clementbalestrat) [inline src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js]: same

File: src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+			} else if (collateralAmount < MIN_LOAN_SIZE) {
+				setCollateralAmountErrorMessage(
+					t('loans.loan-card.errors.min-loan-size', {
+						currencyKey: COLLATERAL_CURRENCY_KEY,
+						minLoanSize: MIN_LOAN_SIZE,
+					})
+				);
+			}
+		}
+	}, [collateralAmount, collateralCurrencyBalance, t, currentWallet]);
```
Comment (clementbalestrat) [inline src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js]: do you need all of these? Might trigger useEffect for no reasons.

If you wanted to silent the eslint error maybe just use `// eslint-disable-next-line react-hooks/exhaustive-deps`

File: src/pages/Root/Root.js [hunk 4]
```diff
@@ -64,6 +64,7 @@ const Root = ({
 	const fetchAndSetWalletBalances = useCallback(
 		async synths => {
 			if (!currentWallet) return;
+			console.log(await getWalletBalances(currentWallet, synths));
```
Comment (clementbalestrat) [inline src/pages/Root/Root.js]: same

File: src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js (hunk 3 above)
Comment (evgenyboxer) [reply src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js]: Yeah all of these are used

File: src/pages/Loans/components/LoanCards/CloseLoanCard/CloseLoanCard.js (hunk 1 above)
Comment (evgenyboxer) [reply src/pages/Loans/components/LoanCards/CloseLoanCard/CloseLoanCard.js]: yeah this is needed for testing

File: src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js (hunk 2 above)
Comment (evgenyboxer) [reply src/pages/Loans/components/LoanCards/CreateLoanCard/CreateLoanCard.js]: for testing ^

File: src/pages/Root/Root.js (hunk 4 above)
Comment (evgenyboxer) [reply src/pages/Root/Root.js]: fixed

**Pull request merged**